Formatting fixes

NavrajBal opened the pull request:
<!-- A quick summary of the implementation details -->
## Giga linting PR 
Made various fixes to adress linter concerns in backend and some in frontend

Many issues were addressed by simply applying prettier.

Other types of problems:
- Incorrect import ordering
- Unnecessary use of await in functions (wrapped in try catch for error handling also adressing this problem)
- Use of await in for of loops (used promise.all() for asynchronous calling)

## Checklist
- [ ] All of my containers are healthy and not producing any error messages
- [ ] The frontend and backend ports are set to 3000 & 5000, respectively
- [ ] If new packages were installed, I ran yarn install, and there is no package-lock.json file present in the codebase

NavrajBal committed:
todolistwidget format and add interfact for assigned task

NavrajBal committed:
small formating fixes and adding of error handling

NavrajBal committed:
move node cron from devdependency to project

NavrajBal committed:
fix linter errors in assignedTaskResolver

NavrajBal committed:
updateLoginStreak formatting fix

NavrajBal committed:
evaluateBadge formatting fix

NavrajBal committed:
badgeResolver formatting and linter fixes

NavrajBal committed:
createAssignedTask formatting fixes

NavrajBal committed:
Replace use of "for of" loop with Promise.all() in creatAssignedTask

NavrajBal committed:
participant resolver formatting and error handling

NavrajBal committed:
login resolver formatting fixes and switch from require to import for jwt module

NavrajBal committed:
used unused variables for error handling and fixed use of any type

NavrajBal committed:
graphql middleware typing fixes

NavrajBal committed:
added reason to update marillac bucks temporary fix

applepie7864 committed:
add back old linter

NavrajBal committed:
remove unused import

NavrajBal committed:
disable linter for seed-snaplet, removed unused imports, and switched to using promis.all

yanxue06 committed:
Merge branch 'dev' into formatting-fixes

yanxue06 commented:
@NavrajBal I think merge conflicts have to be resolved for GitHub actions to trigger

yanxue06 committed:
bump setup-node version

yanxue06 committed:
update yarn lock and resolve ts compilation errors

yanxue06 committed:
generate schema in CI

yanxue06 committed:
always run lints regardless if past steps fail

yanxue06 committed:
Fix linting errors in frontend utils and backend resolvers

yanxue06 committed:
backend linting

yanxue06 committed:
Fix linting errors in resolver files

- Remove return-await and fix formatting in getParticipantById
- Replace any type with explicit type in getParticipantAnnouncements
- Remove unused where variable in getGoalHistoryByParticipant
- Prefix unused reason parameter with underscore in updateMarillacBucks
- Fix prettier formatting issues

yanxue06 committed:
backend linting

yanxue06 committed:
unused types

yanxue06 commented:
ok did some work in this giga pr and its now passing the linting! There were some unused types I think @jjudyyang was working on that were removed in one of the commits just to pass CI, but we can add them back if they were a part of some bigger PR. Other than that, I think we just do some checks that functionality was preserved.

yanxue06 committed:
Merge branch 'dev' into formatting-fixes

NavrajBal committed:
use userAnnouncement for getAnnouncementsByParticipantId

NavrajBal committed:
frontend formating fixes

NavrajBal committed:
Merge branch 'dev' into formatting-fixes

NavrajBal committed:
quick prettier fixes in backend

yanxue06 committed:
Merge branch 'dev' into formatting-fixes

NavrajBal committed:
participant resolver formatting fix

NavrajBal committed:
disable linter and type checking for reportresolver and sendreport email

NavrajBal committed:
fix type errors in generatedatareportscsv

NavrajBal committed:
mutation formatting fix

NavrajBal committed:
disable eslint any check on component.ts

NavrajBal committed:
formatting ran on all front end files

NavrajBal committed:
various frontend formatting and unused imports warnings adressed

NavrajBal committed:
prettier warnings fixed

NavrajBal committed:
useref for prevpined annoucement expandedview

NavrajBal committed:
Merge branch 'dev' into formatting-fixes

NavrajBal committed:
prettier warning fixes

NavrajBal committed:
add generate eslint breakdown script

NavrajBal committed:
remove duplicates in resolvers.ts

NavrajBal commented:
The linter is passing for the backend now with the most recent chagnes from dev merged in.
Although in most files linter errors were fixed for the following files eslint was disabled to some degree:
frontend/src/types/component.ts
frontend/webpack.config.js
backend/resolvers/reportResolver.ts
backend/crons/scripts/sendReportEmails.ts


As for the frontend, all prettier formatting changes and unused imports were fixed, warnings (~180) now are for:
1) for use of type any 
2) unused variables
3) console logs
4) not ehaustive dependencies in useEffects

I also added a shell script (`./generate-eslint-breakdown.sh`) that can be run by going to the to project root, and running in your terminal:
```
./generate-eslint-breakdown.sh  
```

This generates two text files:
eslint-warnings-by-rule-frontend.txt
eslint-warnings-by-rule-backend.txt

This lists out current eslint warnings and errors seperated by type of error. I feel this will be helpful in testing locally and not needing to wait for it run on github.

I have tested most main functions of the app locally and they are working well as of now.

NavrajBal commented:
Since most functionality is still working I am going to merge to dev now.
